### PR TITLE
Migrate federated file sharing to PSR LoggerInterface

### PR DESCRIPTION
**Pull request by Altahrim**
See  #32127 

## Summary
Remove `OCP\ILogger` from `apps/federatedfilesharing` and use Psr\Log\LoggerInterface` instead.

### CODE REVIEW

File: apps/federatedfilesharing/lib/Controller/MountPublicLinkController.php [hunk 1]
```diff
@@ -94,28 +69,22 @@ class MountPublicLinkController extends Controller {
 	 * @param IUserSession $userSession
 	 * @param IClientService $clientService
 	 * @param ICloudIdManager $cloudIdManager
+	 * @param LoggerInterface $logger
```
Comment (come-nc) [inline apps/federatedfilesharing/lib/Controller/MountPublicLinkController.php:72]: You can remove the `@param` phpdoc tags when strong typing is used in the method prototype already.
Just keep `@param string $appName` if you do not strong type it, but I think you can strong type it as it’s the case in other controllers.

File: apps/federatedfilesharing/tests/Controller/MountPublicLinkControllerTest.php [hunk 2]
```diff
@@ -48,48 +48,49 @@
 use OCP\IUserSession;
 use OCP\Share\IManager;
 use OCP\Share\IShare;
+use Psr\Log\LoggerInterface;
 
 class MountPublicLinkControllerTest extends \Test\TestCase {
-	/** @var IContactsManager|\PHPUnit\Framework\MockObject\MockObject */
+	/** @var IContactsManager|MockObject */
```
Comment (come-nc) [inline apps/federatedfilesharing/tests/Controller/MountPublicLinkControllerTest.php:54]: You need to add a `use PHPUnit\Framework\MockObject\MockObject;`

File: apps/federatedfilesharing/tests/Controller/MountPublicLinkControllerTest.php [hunk 3]
```diff
@@ -48,48 +48,49 @@
...
 	protected $contactsManager;
 
-	/** @var  MountPublicLinkController */
+	/** @var MountPublicLinkController */
 	private $controller;
 
-	/** @var  \OCP\IRequest | \PHPUnit\Framework\MockObject\MockObject */
+	/** @var \OCP\IRequest|MockObject */
```
Comment (come-nc) [inline apps/federatedfilesharing/tests/Controller/MountPublicLinkControllerTest.php:60]:
```suggestion
	/** @var IRequest|MockObject */
```

File: apps/federatedfilesharing/lib/Controller/MountPublicLinkController.php (hunk 1 above)
Comment (Altahrim) [reply apps/federatedfilesharing/lib/Controller/MountPublicLinkController.php]: Done! 
I prefer it that way

**Review by come-nc — APPROVED** [review]

**Review by artonge — APPROVED** [review]

**Pull request merged**